datastore/longhorn.go

```diff
@@ -242,6 +287,11 @@ func (s *DataStore) GetCredentialFromSecret(secretName string) (map[string]strin
 	}
 	credentialSecret := make(map[string]string)
 	if secret.Data != nil {
+		if err := s.annotateAWSIAMRole(string(secret.Data[types.AWSIAMRoleArn])); err != nil {
```

Comment:
It seems this kind of on-the-fly update should be avoided, because we will cause repated query operation to do the check. 

Let's see if it's possible to have secret controller to update the annotations on demand and make the resources update in the controller reconcile loop only.

Reply:
A new secret controller added to update the annotations to longhorn manager and replica instance manager Pods.